src/types.ts

```diff
@@ -193,10 +193,11 @@ export interface InitialOptionsTsJest extends Config.InitialOptions {
   globals?: GlobalConfigTsJest
 }
 type TsJestTransformerOptions = TsJestGlobalOptions
-export interface JestConfigWithTsJest extends Partial<Omit<Config.ProjectConfig, 'transform'>> {
+export interface JestConfigWithTsJest extends Partial<Omit<Config.InitialOptions, 'transform' | 'globals'>> {
   transform: {
     [regex: string]: 'ts-jest' | ['ts-jest', TsJestTransformerOptions] | string | [string, Record<string, unknown>]
   }
+  globals?: { [K: string]: unknown }
```

Comment:
This wasn’t actually the change would fix your issue. Your issue was caused by this one https://github.com/kulshekhar/ts-jest/blob/21c553659ceb2463ea1a8758c60c62ce776b269d/src/types.ts#L20